### ISSUE TITLE
Bad Sad error occured when there is a invalid configuration on ballerina.conf

### ISSUE THREAD

**Issue by ramith** [ballerina-platform/ballerina-lang#19369]
**Description:**
I have below configuration in ballerina.conf (generated by helm)
```toml
[idp.cache]
        expirationPeriod=1.8e+06
        size=1000
```
I do following in my ballerina code:
```ballerina
final int cacheExpiryInterval = config:getAsInt("idp.cache.expirationPeriod");
```
Then I get this in bellerina-internal.log

`
[2019-10-11 15:47:44,569] SEVERE {b7a.log.crash} - idp.cache.expirationPeriod does not map to a valid int
java.lang.IllegalArgumentException: idp.cache.expirationPeriod does not map to a valid int
at org.ballerinalang.config.ConfigRegistry.getAsInt(ConfigRegistry.java:248)
at org.ballerinalang.stdlib.config.GetConfig.get(GetConfig.java:49)
at ballerina.config.natives.get(natives.bal:29)
at ballerina.config.config.getAsInt(config.bal:59)
at choreo.idp.___init.choreo_idp_0_1_0__init_(idp:21)
at choreo.api.___init.$moduleInit(api:65535)
at choreo.api.___init.$lambda$$moduleInit$(api)
at choreo.api.___init$$Lambda$6.0000000024349070.apply(Unknown Source)
at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:486)
at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:237)
at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:207)
at org.ballerinalang.jvm.scheduling.Scheduler.start(Scheduler.java:194)
at choreo.api.___init.main(api)
`
There are two issues here:

1. Config API can't parse scientific notation
2. This error should have been on sysout (than in ballerina-internal.log). It makes devOp's day a bad one.

**Steps to reproduce:**
1. Put above code in a ballerina program.
2. configure the ballerina.conf
3. execute the program.
**Affected Versions:**
1.0.1
**OS, DB, other environment details and versions:**
K8, AWS
**Related Issues (optional):**

**Suggested Labels (optional):**
blocker
**Suggested Assignees (optional):**
@chamil321 @hasithaa

**Comment by pubudu91** [comment]
The issue here is TOML doesn't support scientific notation for integers. Therefore when parsing and resolving the configs, it is parsed as a `float`. Since you are attempting to retrieve it as an `int`, the config registry throws an exception. However, this shouldn't have crashed the runtime. Instead, it should've panicked.

**Comment by pubudu91** [comment]
Fixed by #19395. Hence closing this.
